### PR TITLE
Change default config directories so sudo isn't needed

### PR DESCRIPTION
**Pull request by rschlussel**
@anusudarsan is  updating the docs in a separate pr.

### CODE REVIEW

File: prestoadmin/util/local_config_util.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def get_config_directory():
 def get_log_directory():
     config_directory = os.environ.get(LOG_DIR_ENV_VARIABLE)
     if not config_directory:
-        config_directory = DEFAULT_PRESTOADMIN_LOG_DIR
```
Comment (anusudarsan) [inline prestoadmin/util/local_config_util.py]: why did you remove this env variable from constants.py instead of just updating the default there (like you have the DEFAULT_LOCAL_CONF_DIR)?

File: tests/product/standalone/test_installation.py [hunk 2]
```diff
@@ -55,10 +55,17 @@ def test_install_non_root(self):
         """.format(mount_dir=self.cluster.mount_dir,
                    install_dir=install_dir)
 
-        self.assertRaisesRegexp(OSError, 'mkdir: cannot create directory '
-                                '`/var/log/prestoadmin\': Permission denied',
-                                self.cluster.run_script_on_host, script,
-                                self.cluster.master)
+        self.cluster.run_script_on_host(script, self.cluster.master)
```
Comment (petroav) [inline tests/product/standalone/test_installation.py]: So if we're still keeping this test, how are the remaining tests being run? As root? I saw that you set `invoke_sudo` to False. Can you clarify?

File: tests/configurable_cluster.py [hunk 3]
```diff
@@ -211,8 +212,6 @@ def copy_to_host(self, source_path, host, dest_path=None):
             dest_path = os.path.join(self.mount_dir,
                                      os.path.basename(source_path))
         self.exec_cmd_on_host(host, 'mkdir -p {dir}'.format(dir=os.path.dirname(dest_path)))
-        self.exec_cmd_on_host(host, 'chown {user}:{group} {file}'.format(
```
Comment (petroav) [inline tests/configurable_cluster.py]: Why is this no longer needed?

File: tests/product/base_product_case.py [hunk 4]
```diff
@@ -239,7 +239,7 @@ def run_prestoadmin(self, command, raise_error=True, cluster=None,
         command = self.replace_keywords(command, cluster=cluster, **kwargs)
         return cluster.exec_cmd_on_host(
             cluster.master,
-            "/opt/prestoadmin/presto-admin --user {user} {cmd}".format(
+            "~/prestoadmin/presto-admin --user {user} {cmd}".format(
```
Comment (petroav) [inline tests/product/base_product_case.py]: Same comment about parametrizing the location of the config directories.

File: tests/product/base_product_case.py [hunk 5]
```diff
@@ -187,7 +187,7 @@ def dump_and_cp_topology(self, topology, cluster=None):
             cluster = self.cluster
         cluster.write_content_to_host(
             json.dumps(topology),
-            '/etc/opt/prestoadmin/config.json',
+            '~/.prestoadmin/config.json',
```
Comment (petroav) [inline tests/product/base_product_case.py]: Yea you're right, we should haven working commits. Before pushing can you squash product changes with all test changes?

File: tests/configurable_cluster.py [hunk 6]
```diff
@@ -125,21 +128,19 @@ def stop_host(self, host_name):
...
-        )
-        self.exec_cmd_on_host(
-            self.master,
-            'sed -i s/%s/%s/g ~/.prestoadmin/config.json'
-            % (ips[host_name], down_hostname)
-        )
+        self.exec_cmd_on_host(self.master,
+                              'sed -i s/%s/%s/g %s' %
+                              (host_name, down_hostname, get_config_file_path())
+                              )
```
Comment (petroav) [inline tests/configurable_cluster.py]: The closing parenthesis should be more towards the beginning of the line.

File: tests/product/standalone/test_installation.py (hunk 2 above)
Comment (petroav) [inline tests/product/standalone/test_installation.py]: What about EMR, what user are we running with there? cc @anusudarsan

File: tests/product/config_dir_utils.py [hunk 7]
```diff
@@ -20,5 +24,17 @@ def get_connectors_directory():
...
 def get_mode_config_path():
     return os.path.join(get_config_directory(), 'mode.json')
+
+
+def get_install_directory():
+    return os.path.join(os.path.join('~', 'prestoadmin'))
+
+
+def get_presto_admin_path():
+    return os.path.join(os.path.join(get_install_directory(), 'presto-admin'))
```
Comment (petroav) [inline tests/product/config_dir_utils.py]: I think the outer `os.path.join` is not necessary, right?

File: tests/product/config_dir_utils.py [hunk 8]
```diff
@@ -20,5 +24,17 @@ def get_connectors_directory():
...
+def get_log_directory():
+    return os.path.join(get_config_directory(), 'log')
+
+
 def get_mode_config_path():
     return os.path.join(get_config_directory(), 'mode.json')
+
+
+def get_install_directory():
+    return os.path.join(os.path.join('~', 'prestoadmin'))
```
Comment (rschlussel) [inline tests/product/config_dir_utils.py]: nope, thanks

File: tests/configurable_cluster.py (hunk 6 above)
Comment (rschlussel) [inline tests/configurable_cluster.py]: changed it.  (also had to move the first argument down one line to do that, or you get a pep warning about indentation).

**Comment by rschlussel** [comment]
merged
